client/cli/tx.go

```diff
@@ -54,3 +68,114 @@ func NewCmdExecute() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "software-upgrade <name> [--upgrade-height <height>] [--upgrade-info <info>] [flags]",
+		Args:  cobra.ExactArgs(1),
+		Short: "Schedule a software upgrade on chain",
+		Long: "Schedule a software upgrade on chain.\n" +
+			"Please specify a unique name and height for the upgrade to take effect.\n" +
+			"You may include info to reference a binary download link, in a format compatible with: https://docs.cosmos.network/main/build/tooling/cosmovisor",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			clientCtx, err := client.GetClientTxContext(cmd)
+			if err != nil {
+				return err
+			}
+
+			name := args[0]
+			p, err := parsePlan(cmd.Flags(), name)
+			if err != nil {
+				return err
+			}
+
+			noValidate, err := cmd.Flags().GetBool(FlagNoValidate)
+			if err != nil {
+				return err
+			}
+
+			if !noValidate {
+				daemonName, err := cmd.Flags().GetString(FlagDaemonName)
+				if err != nil {
+					return err
+				}
+
+				noChecksum, err := cmd.Flags().GetBool(FlagNoChecksumRequired)
+				if err != nil {
+					return err
+				}
+
+				var planInfo *plan.Info
+				if planInfo, err = plan.ParseInfo(p.Info, plan.ParseOptionEnforceChecksum(!noChecksum)); err != nil {
+					return err
+				}
+
+				if err = planInfo.ValidateFull(daemonName); err != nil {
+					return err
+				}
+			}
+
+			msgExecute := types.NewMsgExecute(
+				clientCtx.FromAddress.String(),
+				[]sdk.Msg{
+					&upgradetypes.MsgSoftwareUpgrade{
+						Authority: types.ModuleAddress.String(),
+						Plan:      p,
+					},
+				})
+
+			return tx.GenerateOrBroadcastTxCLI(clientCtx, cmd.Flags(), msgExecute)
+		},
+	}
+
+	cmd.Flags().Int64(FlagUpgradeHeight, 0, "The height at which the upgrade must happen")
+	cmd.Flags().String(FlagUpgradeInfo, "", "Info for the upgrade plan such as new version download urls, etc.")
+	cmd.Flags().Bool(FlagNoValidate, false, "Skip validation of the upgrade info (dangerous!)")
+	cmd.Flags().Bool(FlagNoChecksumRequired, false, "Skip requirement of checksums for binaries in the upgrade info")
+	cmd.Flags().String(FlagDaemonName, getDefaultDaemonName(), "The name of the executable being upgraded (for upgrade-info validation). Default is the DAEMON_NAME env var if set, or else this executable")
+
+	flags.AddTxFlagsToCmd(cmd)
+
+	return cmd
+}
+
+// newSubmitRecoverClientCmd defines the command to recover an IBC light client.
+// This command has been adapted from ibc-go's `newSubmitRecoverClientProposalCmd`.
+// https://github.com/cosmos/ibc-go/blob/v8.5.2/modules/core/02-client/client/cli/tx.go#L249
+func newSubmitRecoverClientCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "recover-client [subject-client-id] [substitute-client-id] [flags]",
+		Args:  cobra.ExactArgs(2),
+		Short: "recover an IBC client",
+		Long: `recover an IBC client
+		Please specify a subject client identifier you want to recover
+		Please specify the substitute client the subject client will be recovered to.`,
```

Comment:
Same as above, but also, I don't think the long is necessary as they are just expanding on the flags.

```suggestion
		Short: "Helper for recovering an expired client",
```